service/internal/cloudconfig/tcnp.go

```diff
@@ -149,7 +150,19 @@ func (t *TCNP) NewTemplates(ctx context.Context, obj interface{}) ([]string, err
 		// Required for proper rending of the templates.
 		params = k8scloudconfig.DefaultParams()
 
-		params.Cluster = cmaClusterToG8sConfig(t.config, cl, key.KubeletLabelsTCNP(&cr)).Cluster
+		g8sConfig := cmaClusterToG8sConfig(t.config, cl, key.KubeletLabelsTCNP(&cr))
+
+		if cl.Spec.Provider.Pods.CIDRBlock != "" {
+			_, ipnet, err := net.ParseCIDR(cl.Spec.Provider.Pods.CIDRBlock)
+			if err != nil {
+				return []string{""}, microerror.Mask(err)
```

Comment:
When an error is returned we should comply with the convention of returning the zero value of the other return parameters. 

```suggestion
				return nil, microerror.Mask(err)
```

Reply:
right